apps/web/src/pages/quick-start/steps/Setup.tsx

```diff
@@ -0,0 +1,126 @@
+import { Stack, Stepper } from '@mantine/core';
+
+import { ChannelCTATypeEnum, ICreateNotificationTemplateDto, INotificationTemplate, StepTypeEnum } from '@novu/shared';
+
+import { QuickStartWrapper } from '../components/QuickStartWrapper';
+import { useNotificationGroup } from '../../../api/hooks/useNotificationGroup';
+import { useTemplates } from '../../../api/hooks/useTemplates';
+import { useEnvController } from '../../../store/useEnvController';
+import { useInAppActivated } from '../components/useInAppActivated';
+import { APPLICATION_IDENTIFIER, frameworkInstructions, notificationTemplateName } from '../consts';
+import { createTemplate } from '../../../api/notification-templates';
+import { LoaderProceedTernary } from '../components/LoaderProceedTernary';
+import { Prism } from '../../settings/tabs/components/Prism';
+import { When } from '../../../components/utils/When';
+import { colors } from '../../../design-system';
+
+export function Setup() {
+  const [notificationTemplate, setNotificationTemplate] = useState<INotificationTemplate>();
+  const { framework } = useParams();
+  const { groups } = useNotificationGroup();
+  const { templates = [], loading } = useTemplates();
+  const { environment } = useEnvController();
+  const { initialized } = useInAppActivated();
+
+  const instructions = frameworkInstructions.find((instruction) => instruction.key === framework)?.value ?? [];
+  const openBrowser = framework === 'demo';
+
+  const environmentIdentifier = environment?.identifier ? environment.identifier : '';
+
+  const { mutateAsync: createNotificationTemplate } = useMutation<
+    INotificationTemplate,
+    { error: string; message: string; statusCode: number },
+    ICreateNotificationTemplateDto
+  >(createTemplate);
+
+  useEffect(() => {
+    if (!loading) {
+      const onboardingNotificationTemplate = templates.find((template) =>
+        template.name.includes(notificationTemplateName)
+      );
+
+      if (onboardingNotificationTemplate) {
+        setNotificationTemplate(onboardingNotificationTemplate);
+      } else {
+        createOnBoardingTemplate();
+      }
+    }
+  }, [templates]);
+
+  async function createOnBoardingTemplate() {
+    const payloadToCreate = {
+      notificationGroupId: groups[0]._id,
+      name: notificationTemplateName,
+      active: true,
+      draft: false,
+      steps: [
+        {
+          template: {
+            type: StepTypeEnum.IN_APP,
+            content:
+              'Welcome to Novu! Click on this notification to <b>visit the cloud admin panel</b> managing this message',
+            cta: {
+              type: ChannelCTATypeEnum.REDIRECT,
+              data: { url: `/templates/edit/${notificationTemplate?._id}` },
+            },
+          },
+        },
+      ],
+    } as ICreateNotificationTemplateDto;
+
+    const createdTemplate = await createNotificationTemplate(payloadToCreate);
+    setNotificationTemplate(createdTemplate);
```

Comment:
Can we only create it once if it's not already exists?

Reply:
Do you mean to reuse old ones like the CLI one? i think i asked if we will want to reuse it in the end or not. Let me know if yes. @scopsy 
If you mean on this flow, then yes there is a validation before the call to this function. 

```
      const onboardingNotificationTemplate = templates.find((template) =>
        template.name.includes(notificationTemplateName)
      );

      if (onboardingNotificationTemplate) {
        setNotificationTemplate(onboardingNotificationTemplate);
      } else {
        createOnBoardingTemplate();
      }
```

the `notificationTemplateName` is a predefined template name. (first time create second-time skip)

Reply:
oh got ya. That makes sense